pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/dao/stoolupload/StoolUploadDao.java

```diff
@@ -0,0 +1,95 @@
+import org.broadinstitute.dsm.util.DSMConfig;
+import org.broadinstitute.dsm.util.EventUtil;
+import org.broadinstitute.lddp.db.SimpleResult;
+
+public class StoolUploadDao implements Dao<StoolUploadDto> {
+    private static final String SELECT_KIT_ID = "SELECT dsm_kit_id FROM ddp_kit INNER JOIN ddp_kit_request dkr"
+            + " on ddp_kit.dsm_kit_request_id = dkr.dsm_kit_request_id";
+    private static final String BY_BARCODE = " WHERE kit_label = ?";
+    private static final String BY_PT_ID = " dkr.ddp_participant_id = ?";
+    private static final String UPDATE_KIT = "UPDATE ddp_kit SET receive_date = ?, receive_by = 'HSPH'";
+    private static final String BY_KIT_ID = " WHERE dsm_kit_id = ?";
+    private static final String IF_NULL = " AND receive_date IS NULL";
+
+    public Optional<StoolUploadDto> getStoolUploadDto(String participantId, String mfBarcode) {
+        SimpleResult simpleResult = TransactionWrapper.inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult(0);
+            try (PreparedStatement stmt = conn.prepareStatement(SELECT_KIT_ID + BY_BARCODE + " AND " + BY_PT_ID)) {
+                stmt.setString(1, mfBarcode);
+                stmt.setString(2, participantId);
+                try (ResultSet rs = stmt.executeQuery()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = new StoolUploadDto(rs.getInt(DBConstants.DSM_KIT_ID));
+                    } else {
+                        throw new RuntimeException("No kit found with the barcode " + mfBarcode);
+                    }
+                }
+            } catch (SQLException e) {
+                dbVals.resultException = e;
+            }
+            return dbVals;
+        });
+        if (simpleResult.resultException != null) {
+            throw new RuntimeException("No kit found with the barcode " + mfBarcode,
+                    simpleResult.resultException);
+        }
+        return Optional.of((StoolUploadDto) simpleResult.resultValue);
+    }
+
+    public void updateKitData(String receiveDate, int kitId, String mfBarcode) {
+        SimpleResult simpleResult = TransactionWrapper.inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult(0);
+            try (PreparedStatement stmt = conn.prepareStatement(UPDATE_KIT + BY_KIT_ID + IF_NULL)) {
+                stmt.setString(1, receiveDate);
+                stmt.setInt(2, kitId);
+                int updateAmount = stmt.executeUpdate();
+
+                if (updateAmount == 1) {
+                    KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(
+                            DSMConfig.getSqlFromConfig(ApplicationConfigConstants.GET_RECEIVED_KIT_INFORMATION_FOR_NOTIFICATION_EMAIL),
+                            mfBarcode, 1);
+
+                    if (kitDDPNotification != null) {
+                        EventUtil.triggerDDP(conn, kitDDPNotification);
+                    }
+                }
```

Comment:
Optional: We can add `else` statement in case `updateAmount` is not equal to 1 and throw RuntimeException indicating that there was an error while updating kit, it will help us tracking down the possible errors faster.

P.S We can also add successful logs in case everything goes well. For reference please take a look at `DDPInstitutionDao` line 57.